kinto/views/admin.py

```diff
@@ -0,0 +1,168 @@
+                # It's being deleted already.
+                include = False
+                break
+            parent_uri = get_parent_uri(parent_uri)
+
+        if include:
+            ancestor_object_uris.add(object_uri)
+
+    return list(ancestor_object_uris)
+
+@resource.register(name='user-data',
+                   description='Delete the data owned by a user',
+                   plural_path='/__user_data__',
+                   object_path='/__user_data__/{{principal}}',
+                   viewset=UserDataViewSet(),
+                   collection_methods=('DELETE',))
+class UserData(resource.ShareableResource):
+
+    schema = Deleted
+
+    def delete(self):
+        principal = self.request.matchdict['principal']
+        storage = self.request.registry.storage
+        permission = self.request.registry.permission
+        object_uris_and_permissions = permission.get_accessible_objects([principal])
+        object_uris = list(object_uris_and_permissions.keys())
+        write_perm_principals = permission.get_objects_permissions(object_uris, ['write'])
+        to_delete = []
+        for object_uri, principals in zip(object_uris, write_perm_principals):
+            if 'write' not in principals:
+                continue
+            principals = principals['write']
+            # "Ownership" isn't a real concept in Kinto, so instead we
+            # define ownership as meaning "this user is the only one
+            # who can write to this object".
+            if principals == set([principal]):
+                to_delete.append(object_uri)
+
+        print("Going to delete", to_delete)
+        to_delete = condense_under_parents(self.request, to_delete)
+        print("Condensed into", to_delete)
+
+        # Group by (parent_uri, resource of child) to make fewer
+        # requests to storage backend.
+        objects_by_parent_uri = collections.defaultdict(list)
+        # We have to get the matchdict of the child here anyhow, so
+        # keep that to generate events later.
+        matchdicts_by_parent_uri = {}
+        for object_uri in to_delete:
+            parent_uri = get_parent_uri(object_uri)
+            resource_name, matchdict = core_utils.view_lookup(self.request,
+                                                              object_uri)
+            objects_by_parent_uri[(parent_uri, resource_name)].append(matchdict['id'])
+            # This overwrites previous matchdicts for the parent, but
+            # we'll only use the fields that are relevant to the
+            # parent, which will be the same for each child.
+            matchdicts_by_parent_uri[parent_uri] = matchdict
+
+        for (parent_uri, resource_name), object_ids in objects_by_parent_uri.items():
+            # Generate the parent matchdict from an arbitrary child's matchdict.
+            matchdict = {**matchdicts_by_parent_uri[parent_uri]}
+            matchdict.pop('id', None)
+
+            # FIXME: Deletes are paginated too.. ugh
```

Comment:
Why is that?

I think what limits storage operations is this setting https://github.com/Kinto/kinto/blob/f1da55010ae7fe4a4be7d2c8378a35d82eecf84c/kinto/core/__init__.py#L92

Reply:
That's it, thanks. I was trying to figure out a way to get access to it rather than hard-coding a random number..